client/components/vertical-nav/item/index.jsx

```diff
@@ -67,8 +68,14 @@ class VerticalNavItem extends Component {
 
 		const linkProps = external ? { target: '_blank', rel: 'noreferrer' } : {};
 
+		const pathObj = disabled
```

Comment:
I wanted to use a better approach here.

Instead of having a path object and spreading it, I wanted to do something like:

```suggestion
		const WrapperElem = disabled ? 'span' : 'a';
		
		return <WrapperElem href={ path } onClick={ onClick } { ...linkProps }>
```
Benefits include:
1. No need to disable any eslint errors
2. The `href` attribute would automatically be ignored.

I had to decide against it because some folks might use something like:
```CSS
.vertical-nav a { /* some CSS */ }
``` 

Yay or nay?

Reply:
I think we can be more flexible with the new prop, as no existing use cases depend on it. So I do like the approach of using a `span` when we don't actually have a clickable link. We might actually drop the wrapper altogether, as it only serves as a way to intercept user interactions/clicks. Removing the `a` also removes any link-specific behaviour for the element -- I currently see a hover colour change, for example.

 I am thinking something like the following:
```suggestion
		const navItemCard = (
			<CompactCard className={ compactCardClassNames }>
				{ this.getIcon() }
				<span>{ children }</span>
			</CompactCard>;
		);
		
		// If the nav item is disabled, don't wrap it in an <a>, just return the unwrapped card.
		if ( disabled ) {
			return navItemCard;
		}
		
		return (
			<a href={ path } onClick={ onClick } { ...linkProps }>
				{ navItemCard }
			</a>
		);
```

**However**, I think we should make sure document that in the README to make it clear that disabled items are not wrapped by an `a` element, as that is a non-obvious side-effect of the property.